sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/EnvironmentResource.java

```diff
@@ -7,21 +7,23 @@
 
 import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.api.common.AttributesBuilder;
-import io.opentelemetry.sdk.resources.ResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
 
-public class EnvironmentResource extends ResourceProvider {
+final class EnvironmentResource {
```

Comment:
shouldn't this be public so that users who want to do everything manually can get a handle to it?

Reply:
(just to make it like the other resources available in the resource module)

Reply:
We can do that later

Reply:
If this will be left public, add final please :)

Reply:
#2858